### DISCUSSION TITLE
Is it possible to use rewrite params to create a dynamic url instead of just the path

### DISCUSSION THREAD

**Discussion by Angusross111** [vercel/next.js#43353] [Help] [answered]
I currently have multiple API endpoints I want to query from the one page based off an ID:
```
https://id1.domain.com:1234
https://id2.domain.com:1234
https://id3.domain.com:1234
```
Is it possible to do this with the rewrites function in the `next.config.js`. For example:
```
{
    source: "/api/:id",
    destination: "https://:id.domain.com:1234",
},
```

**Comment by icyJoseph** [comment, marked as answer]
I tried this:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return {
      beforeFiles: [
        { source: "/api/:id", destination: "https://:id.example.com" },
      ],
    };
  },
};

module.exports = nextConfig;
```

And then when navigating to http://localhost:3000/api/1234, it tries to actually navigate to `https://:id.example.com` (and fails because it doesn't exist)

If I make a more useful example:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return {
      beforeFiles: [
        {
          source: "/api/:name/:version/:pokemon",
          destination: "https://:name.co/api/:version/pokemon/:pokemon",
        },
      ],
    };
  },
};

module.exports = nextConfig;

```

And then go to `http://localhost:3000/api/pokeapi/v2/151` I get the JSON for the 151th Pokémon, because the request turns into: `https://pokeapi.co/api/v2/pokemon/151`

**Reply by Angusross111** [reply]
Thanks! I think the problem I was having was missing the trailing slash
```
destination: "https://:id.domain.com:1234/",
```
after the port that was throwing the error. Although it was a different Error to what I had 2 weeks ago. But its work now anyway
